### PR TITLE
chore: unlink due date and received date widget

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
Zenhub: [721](https://github.com/bcgov/cas-cif/issues/721) and [748](https://github.com/bcgov/cas-cif/issues/748)

### CODE REVIEW

File: app/tests/unit/lib/theme/widgets/ReceivedDateWidget.test.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe("The ReceivedDateWidget", () => {
       required: true,
     };
     render(<ReceivedDateWidget {...props} />);
-    expect(screen.getByText(/due in 52 weeks/i)).toBeInTheDocument();
+    expect(screen.getByText(/Select a date/i)).toBeInTheDocument();
   });
 
   it("displays overdue text and icon if report is late", () => {
```
Comment (matthieu-foucault) [inline app/tests/unit/lib/theme/widgets/ReceivedDateWidget.test.tsx:65]: this whole test should be removed

File: app/lib/theme/widgets/ReceivedDateWidget.tsx [hunk 2]
```diff
@@ -3,75 +3,34 @@ import React, { forwardRef, useMemo, useRef } from "react";
...
-        : `Due in ${daysFromToday} ${
-            daysFromToday === 1 ? "day" : "days"
-          } (${formattedDueDate})`;
-    }, [formContext?.dueDate, value]);
+      return (
+        <>
+          <div className="receivedStringWrapper">
+            <span style={{ marginRight: "1em" }}>Received</span>
+            <FontAwesomeIcon icon={faCheck} color={"green"} />
+          </div>
```
Comment (matthieu-foucault) [inline app/lib/theme/widgets/ReceivedDateWidget.tsx:23]: Other than this, which could be a `contentPrefix` `uiSchema` property, this is a simple date widget now, right?

File: app/lib/theme/widgets/ReceivedDateWidget.tsx [hunk 3]
```diff
@@ -3,75 +3,34 @@ import React, { forwardRef, useMemo, useRef } from "react";
...
+          ({formattedReceivedDate})
+        </>
+      );
+    }, [value]);
 
     return (
       <div className="receivedDateWrapper">
         <div onClick={onClick} ref={ref} aria-label={label}>
-          {displayString}
+          {value ? displayString : "Select a date"}
```
Comment (matthieu-foucault) [inline app/lib/theme/widgets/ReceivedDateWidget.tsx:32]: Do we even want to display "Select a date" ? If we do, should it be greyed out, at it's a placeholder?

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: app/tests/unit/lib/theme/widgets/DateWidget.test.tsx [hunk 4]
```diff
@@ -41,53 +43,26 @@ describe("The ReceivedDateWidget", () => {
     expect(handleOnChange).toHaveBeenCalledWith(expectedDate);
   });
 
-  it("if report is upcoming, displays weeks until due", () => {
-    const dueDate = getTimestamptzFromDate(
-      DateTime.now().setZone("America/Vancouver").plus({ years: 1 }),
-      true
-    );
+  // it("if report is upcoming, displays weeks until due", () => {
```
Comment (matthieu-foucault) [inline app/tests/unit/lib/theme/widgets/DateWidget.test.tsx:46]: delete commented out code

File: app/tests/unit/lib/theme/widgets/DateWidget.test.tsx [hunk 5]
```diff
@@ -99,8 +74,17 @@ describe("The ReceivedDateWidget", () => {
...
+          <div>
+            <span style={{ marginRight: "1em" }}>Received</span>
+            <FontAwesomeIcon icon={faCheck} color={"green"} />
+          </div>
+        ),
+      },
     };
-    const { container } = render(<ReceivedDateWidget {...props} />);
+    const { container } = render(<DateWidget {...props} />);
+    screen.logTestingPlaygroundURL();
```
Comment (matthieu-foucault) [inline app/tests/unit/lib/theme/widgets/DateWidget.test.tsx:87]:
```suggestion
```

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
just a little bit of cleanup needed in some test code

File: app/tests/unit/lib/theme/widgets/DateWidget.test.tsx (hunk 4 above)
Comment (Sepehr-Sobhani) [reply app/tests/unit/lib/theme/widgets/DateWidget.test.tsx:46]: Oops 🤦‍♂️

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**